### ISSUE TITLE
`p_group_cohomology` build failure

### ISSUE THREAD

**Issue by beew** [sagemath/sage#36821]
### Steps To Reproduce

I tried to compile sage 10.2 and failed at several packages involving cython errors

Compilation failed at packages cylp and p_group_cohomology (maybe more, but I am not continuing for now)
 
Both failures involve error in compiling cython files.

I attach the error logs
[cylp-0.91.5.log](https://github.com/sagemath/sage/files/13575284/cylp-0.91.5.log)
[p_group_cohomology-3.3.3.p1.log](https://github.com/sagemath/sage/files/13575291/p_group_cohomology-3.3.3.p1.log)

### Expected Behavior

compilation suceeds 

### Actual Behavior

Build failed.

### Additional Information

Download the release tarball from github, run bootstrap and then configure (--enable packages) then make -j8

### Environment

```markdown
- **OS**: Ubuntu 22.04
- **Sage Version**:10.2
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

**Comment by dimpase** [comment]
`[p_group_cohomology](https://github.com/sagemath/sage/files/13575291/p_group_cohomology-3.3.3.p1.log)` is broken, it returns wrong results :-( - that's why noone updated it I suppose

**Comment by dimpase** [comment]
cylp's version needs to be updated to 0.92.2 (it builds with Cython 3)

**Comment by mkoeppe** [comment]
Yes, these are known problems in 10.2; see https://github.com/sagemath/sage/wiki/Sage-10.2-Release-Tour#known-problems-and-workarounds

**Comment by mkoeppe** [comment]
> cylp's version needs to be updated to 0.92.2 (it builds with Cython 3)

That's #36439, which has been waiting for review for quite a while now.

**Comment by dimpase** [comment]
> > cylp's version needs to be updated to 0.92.2 (it builds with Cython 3)
> 
> That's #36439, which has been waiting for review for quite a while now.

the latter works on Linux, testing on macOS atm
